packages/react-components/react-avatar/src/utils/getPartitionedAvatarGroupItems.ts

```diff
@@ -0,0 +1,21 @@
+/**
+ * Get the inline items and overflowing items based on the array of AvatarGroupItems needed for AvatarGroup.
+ *
+ * @param items - Array of AvatarGroupItems
+ * @param maxItems - Maximum number of items to show inline
+ *
+ * @returns Two arrays of AvatarGroupItems splitted between the inline items and the overflowing items
+ * based on maxItems.
+ *
+ * NOTE: Each item in the array must be a single AvatarGroupItem. If an item in the array contains a fragment with
+ * two avatars, the items will not be partitioned correctly.
+ */
+export const getPartitionedAvatarGroupItems = (items: JSX.Element[], maxItems = 5) => {
```

Comment:
I was planning on adding the pie portion later since I'm trying to keep this PR as short as possible, but I do agree getting the overall picture in this case is better.